sync:
  image: n3llyb0y/wait
  links:
    - auth
    - call-logd
    - confd
    - postgres
    - rabbitmq
  environment:
    PORTS: "5432 5672 9298 9486 9497"
    TIMEOUT: "${INTEGRATION_TEST_TIMEOUT}"

auth:
  image: p0bailey/docker-flask
  ports:
    - "9497"
  volumes:
    - "./xivo-auth:/usr/local/share/xivo-auth"
    - "./ssl:/usr/local/share/ssl"
  command: "python /usr/local/share/xivo-auth/mock-xivo-auth.py 9497"

call-logd:
  image: xivo-call-logd-test
  volumes:
    - "../../..:/usr/src/xivo-call-logs"
    - "./etc/xivo-call-logd:/etc/xivo-call-logd"
    - "./ssl:/usr/local/share/ssl"
    # - "${LOCAL_GIT_REPOS}/xivo-lib-python/xivo:/usr/local/lib/python2.7/site-packages/xivo"
    # - "${LOCAL_GIT_REPOS}/xivo-dao/xivo_dao:/usr/local/lib/python2.7/site-packages/xivo_dao"
  ports:
    - "9298"
  links:
    - auth
    - confd
    - postgres
    - rabbitmq

confd:
  image: p0bailey/docker-flask
  ports:
    - "9486"
  volumes:
    - "./xivo-confd:/usr/local/share/xivo-confd"
    - "./ssl:/usr/local/share/ssl"
  command: "python /usr/local/share/xivo-confd/mock-xivo-confd.py 9486"

postgres:
  image: wazopbx/postgres-test
  ports:
    - "5432"

rabbitmq:
  image: rabbitmq
  ports:
    - "5672"
